lib/ansible/plugins/connection/aws_ssm.py

```diff
@@ -0,0 +1,535 @@
+            if self._poll_stdout.poll(1000):
+                line = self._filter_ansi(self._stdout.readline())
+                display.vvvv(u"EXEC stdout line: {0}".format(to_text(line)), host=self.host)
+            else:
+                display.vvvv(u"EXEC remaining: {0}".format(remaining), host=self.host)
+                continue
+
+            if mark_start in line:
+                begin = True
+                if not line.startswith(mark_start):
+                    stdout = ''
+                continue
+            if begin:
+                if mark_end in line:
+                    for line in stdout.splitlines():
+                        display.vvvvv(u"POST_PROCESS: {0}".format(to_text(line)), host=self.host)
+                    returncode, stdout = self._post_process(stdout)
+                    break
+                else:
+                    stdout = stdout + line
+
+        stderr = self._flush_stderr(session)
+
+        return (returncode, stdout, stderr)
+
+    def _prepare_terminal(self):
+        ''' perform any one-time terminal settings '''
+
+        if not self.is_windows:
+            cmd = "stty -echo\n" + "PS1=''\n"
+            if PY3:
+                cmd = to_bytes(cmd)
+            self._session.stdin.write(cmd)
+
+    def _wrap_command(self, cmd, sudoable, mark_start, mark_end):
+        ''' wrap command so stdout and status can be extracted '''
+
+        if self.is_windows:
+            cmd = cmd + "; echo " + mark_start + " $? $LASTEXITCODE\necho " + mark_end + "\n"
+            time.sleep(1)
+        else:
+            if sudoable:
+                cmd = "sudo " + cmd
+            cmd = "echo " + mark_start + "\n" + cmd + "\necho $'\\n'$?\n" + "echo " + mark_end + "\n"
+
+        display.vvvv(u"_wrap_command: '{0}'".format(to_text(cmd)), host=self.host)
+        return cmd
+
+    def _post_process(self, stdout):
+        ''' extract command status and strip unwanted lines '''
+
+        if self.is_windows:
+            success = stdout.rfind('True')
+            fail = stdout.rfind('False')
+
+            if success > fail:
+                returncode = 0
+                stdout = stdout[:success]
+            elif fail > success:
+                try:
+                    # test using: ansible -m raw -a 'cmd /c exit 99'
+                    returncode = int(stdout[fail:].split()[1])
+                except (IndexError, ValueError):
+                    returncode = -1
+                stdout = stdout[:fail]
+            else:
+                returncode = -51
+
+            return (returncode, stdout)
+        else:
+            # Get command return code
+            returncode = int(stdout.splitlines()[-2])
+
+            # Throw away ending lines
+            for x in range(0, 3):
+                stdout = stdout[:stdout.rfind('\n')]
+
+            return (returncode, stdout)
+
+    def _filter_ansi(self, line):
+        ''' remove any ANSI terminal control codes '''
+        line = line.decode('utf-8')
```

Comment:
Use `to_text(line)` instead.

Reply:
Updated in commit https://github.com/ansible/ansible/commit/8a5f982ce77b02d747c2f1954c3e413a449211f8

Reply:
Updated in commit https://github.com/ansible/ansible/pull/49652/commits/12c48415bfe8ccab4b4ccb78d3cfb96fc4a93767